configure/lang/go/README.md

```diff
@@ -0,0 +1,96 @@
+# lang-go
+
+This folder contains the deployment manifests for the [Go language extension](https://sourcegraph.com/extensions/sourcegraph/go). 
+
+## Installation instructions
+
+### Setup TLS/SSL 
+
+TLS/SSL is required for secure communication with the language server. Once you have completed ["Configure TLS/SSL"](../../../docs/configure.md#configure-tlsssl) in [docs/configure.md](../../../docs/configure.md#configure-tlsssl), for your overall Sourcegraph instance, you'll need to configure TLS/SSL for the Go language server as well.  
+
+The Go language server needs it's own domain (e.g. `go.sourcegraph.example.com`), and an SSL certificate/key for that domain.
+
+1. Create a [TLS secret](https://kubernetes.io/docs/concepts/configuration/secret/) that contains your TLS certificate and private key for the Go language server.
+
+   ```bash
+   kubectl create secret tls go-tls --key $PATH_TO_KEY --cert $PATH_TO_CERT
+   ```
+
+   Update [create-new-cluster.sh](../../../create-new-cluster.sh) with the previous command.
+
+   ```bash
+   echo kubectl create secret tls go-tls --key $PATH_TO_KEY --cert $PATH_TO_CERT >> create-new-cluster.sh
+   ```
+
+1. Add the above TLS configuration to [configure/lang/go/lang-go.Ingress.yaml](lang-go.Ingress.yaml), making sure to use the real domain name that you are using for the Go language server.
+
+    ```yaml
+    spec:
+        tls:
+        - hosts:
+        # 🚨 TLS is required for secure communication with the language server. 
+        # See the customization guide (../../../docs/configure.md) for information
+        # about configuring TLS
+        #
+        # Make sure to replace 'go.sourcegraph.example.com' with the real domain that you are
+        # using for the Go language server.
+        - go.sourcegraph.example.com
+        secretName: go-tls
+        rules:
+        - host: go.sourcegraph.example.com
+    ```
+
+**WARNING:** Do NOT commit the actual TLS cert and key files to your fork (unless your fork is
+private **and** you are okay with storing secrets in it).
+
+### HTTP basic authentication
+
+HTTP basic authentication is used to prevent unauthorized access to the language server. At a high level, you'll create a secret then put it in both [configure/lang/go/lang-go.Ingress.yaml](lang-go.Ingress.yaml) and in your Sourcegraph global settings so that logged-in users are authenticated when their browser makes requests to the Go language server.
```

Comment:
This does mean that if that basic auth credential is exposed by any Sourcegraph user, anyone with that credential now has unauthorized access to the language server and the code it operates on. Shouldn't we warn people about that? i.e. basic auth is better than nothing, but really you need an actual auth proxy secure your langservers.

Reply:
@slimsag 
This is possible, but this security model is the same as what we currently have as a result of https://github.com/sourcegraph/sourcegraph/issues/2595. We should have a discussion about whether or not this is sufficient, but that shouldn't block this PR.

Reply:
Note that this PR means that part of https://github.com/sourcegraph/sourcegraph/issues/2595#issuecomment-470014851 is now invalid

Reply:
Filed https://github.com/sourcegraph/security-issues/issues/16